packages/maker.js/src/core/chain.ts

```diff
@@ -630,8 +676,35 @@ namespace MakerJs.chain {
      * @param maxArcFacet The maximum length between points on an arc or circle.
      * @returns Array of points which are on the chain.
      */
-    export function toKeyPoints(chainContext: IChain, maxArcFacet?: number): IPoint[] {
-        var result: IPoint[] = [];
+    export function toKeyPoints(chainContext: IChain, maxArcFacet?: number): IPoint[];
```

Comment:
This doesn't seem to be called by the `childrenOnChain` call below. Is it needed to change this?

Reply:
No it’s not referenced in `childrenOnChain` but `toKeyPoints` and `toPoints` were same type of methods and I thought they both can use the callback functionality with path information. If you don’t see it as useful functionality I will remove it.

Reply:
If we make the other changes I've noted, then we won't have the information to mirror the new functionality. So let's back this out.